### PR TITLE
Write a basic boilerplate event bus implementation

### PR DESCRIPTION
**Pull request by petereast**

### CODE REVIEW

**Comment by petereast** [comment]
Couple of things left to do:
- [x] define the interface
- [x] implement a rabbitmq event bus
- [ ] implement a mock event bus

File: message-bus-tester/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "message-bus-tester",
+  "version": "1.0.0",
+  "description": "Eventually this is going to turn into our message bus library, which will eventually implement the pubsweet message bus interface.",
+  "main": "dist/index.js",
+  "scripts": {
+    "test": "jeset",
+    "compile": "tsc",
+    "start:dev": "nodemon --watch 'src/**/*.ts' --ignore 'src/**/*.spec.ts' --exec 'ts-node' src/main.ts"
+  },
+  "author": "Libero Team",
```
Comment (giorgiosironi) [inline message-bus-tester/package.json]: `Libero Reviewer Team` :sweat_smile:

File: message-bus-tester/src/rabbit-message-queue/index.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
+import { EventIdentifier, MessageQueue, Event } from "../message-queue";
+import * as amqplib from "amqplib";
+import { Connection, Message } from "amqplib";
+import { Option, Some } from "funfix";
+import { InfraLogger as logger } from "../logger";
+
+export class RabbitMessageQueue implements MessageQueue {
```
Comment (diversemix) [inline message-bus-tester/src/rabbit-message-queue/index.ts]: Would it be good to split the queue implementation into `Publisher` and `Listener` ?

Comment (petereast) [reply message-bus-tester/src/rabbit-message-queue/index.ts]: :thinking:

Comment (petereast) [reply message-bus-tester/src/rabbit-message-queue/index.ts]: I don't know if it makes sense to split the implementation but I think it makes sense to split the interfaces into `Publisher` and `Subscriber` and maybe `EventBus`

**Comment by petereast** [comment]
Worth noting: This doesn't currently run in CI because our CI in this repo isn't quite setup for this use case, eventually the message bus code will be moved out into it's own repo where it'll be published to npm with it's own pipeline

**Pull request merged**